packages/data-store/src/entities/presentationDefinition/PresentationDefinitionItemEntity.ts

```diff
@@ -0,0 +1,40 @@
+import { BaseEntity, BeforeInsert, BeforeUpdate, Column, CreateDateColumn, Entity, Index, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm'
+import { IsNotEmpty } from 'class-validator'
+
+@Entity('PresentationDefinitionItem')
+@Index(['version'], { unique: false })
+export class PresentationDefinitionItemEntity extends BaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id!: string
+
+  @Column({ name: 'definition_id', length: 255, nullable: false, unique: false })
+  @IsNotEmpty({ message: 'definitionId field must not be empty' })
+  definitionId!: string
+
+  @Column({ name: 'version', length: 255, nullable: false, unique: false })
+  @IsNotEmpty({ message: 'version field must not be empty' })
+  version!: string
+
+  @Column({ name: 'tenant_id', length: 255, nullable: true, unique: false })
+  tenantId?: string
+
+  @Column({ name: 'purpose', length: 255, nullable: true, unique: false })
```

Comment:
It's easier to show in a screen / listview. Also you can filter on it / sort on it in the DB query.

Reply:
Yes we denormalize things like purpose, name, description because we cannot look into every single PD at scale from a list view.
The name of the RP should come from a RP instance we should defined in the future. That will be associated with an identifier, which in turn will be associated with a contact known in the system. For all intents and purposes we should create relationships in the near future between PDs and identifiers acting as RP

Reply:
not sure if i get it. but ik already do things like defintion.name to look into the pd for the name. for the listview you can just look into objects anyway you like

Reply:
I rather have them denormalized. PE will go in the future anyway (to be replaced by another query language, jay)

Reply:
denormalized name also now